src/drt/src/pa/FlexPA_unique.h

```diff
@@ -74,19 +76,73 @@ class UniqueInsts
 
   frDesign* getDesign() const { return design_; }
   frTechObject* getTech() const { return design_->getTech(); }
+
+  /**
+   * @brief Checks if any terminal has a NonDefaultRule.
+   *
+   * @param inst A cell instance.
+   *
+   * @return If instance contains a NonDefaultRule net connected to any
+   * terminal.
+   */
   bool isNDRInst(frInst& inst);
   bool hasTrackPattern(frTrackPattern* tp, const Rect& box) const;
 
+  /**
+   * @brief Creates a vector of preferred track patterns.
+   *
+   * Not every track pattern is a preferred one,
+   * this function acts as filter of design_->getTopBlock()->getTrackPatterns()
+   * to only take the preferred ones.
+   *
+   * @return A vector of track patterns objects.
```

Comment:
returns void, I assume you are planning to make it return the vector directly. if so, just add a TODO comment.

Reply:
Yeah, sorry for the confusion. I am taking the parts of PR [#5744](https://github.com/The-OpenROAD-Project/OpenROAD/pull/5744/files) and making separate PRs for each for easier reviewing. This comment assumes that the changes from PR [#5778](https://github.com/The-OpenROAD-Project/OpenROAD/pull/5778) are already in place, that's why the comment refers to a return and does not mentions an input. As both PRs are still unmerged I will transfer these comments to that PR so its more coherent. Sorry for this oversight